src/tests/test_core_python.py

```diff
@@ -1,70 +1,99 @@
+from functools import cache
 from pathlib import Path
+from typing import Any
 
 import pytest
+import yaml
+from yaml import CLoader as Loader
 
 from pyodide_build.common import UNVENDORED_STDLIB_MODULES
 
 
-def test_cpython_core(python_test, selenium, request):
-
-    name, error_flags = python_test
-
+def filter_info(info: dict[str, Any], browser: str) -> dict[str, Any]:
+    info = dict(info)
     # keep only flags related to the current browser
     flags_to_remove = ["firefox", "chrome", "node"]
-    flags_to_remove.remove(selenium.browser)
-    for flag in flags_to_remove:
-        if "crash-" + flag in error_flags:
-            error_flags.remove("crash-" + flag)
+    flags_to_remove.remove(browser)
+    for browser in flags_to_remove:
+        for key in list(info.keys()):
+            if key.endswith(browser):
+                del info[key]
+    return info
 
-    if any(flag.startswith("segfault") for flag in error_flags):
-        pytest.skip('known segfault: "{}"'.format(" ".join(error_flags)))
 
-    if error_flags:
+def possibly_skip_test(request, info: dict[str, Any]) -> dict[str, Any]:
+    for reason in (
+        reason for [flag, reason] in info.items() if flag.startswith("segfault")
```

Comment:
For tuple unpacking, it's usually a tuple not a list,
```suggestion
        reason for (flag, reason) in info.items() if flag.startswith("segfault")
```